syntax = "proto3";

package dg.model;

import "common.proto";
import "system.proto";

service SkynetService { 
    rpc VideoRecognize(SkynetRequest) returns (SkynetResponse) {}
}

message SkynetRequest {
    SkynetRequestContext Context = 1;
    SkynetRequestData Data = 2;
}

message SkynetResponse {
    SkynetResponseContext Context = 1;
    SkynetResponseData Data = 2;
}

message SkynetRequestContext {
    string SessionId = 1;
    RecognizeType Type = 2;
    StorageConfig Storage = 3;
    map<string, string> Params = 4;
}

message SkynetOutput {
    string Address = 1;
    string Type = 2;
    int32 Bitrate = 3;
    int32 Fps = 4;
    int32 Width = 5;
    int32 Height = 6;
    bool ShotHotspot = 7;
    int32 SnapWidth = 8;
}

message SkynetRequestData {
    int32 SensorId = 1;
    string SensorName = 2;
    string SensorURI = 3;
    string SensorType = 4;
    int32 Width = 5;
    int32 Height = 6;
    int32 Fps = 7;
    repeated Cutboard Hotspots = 8;
    repeated Cutboard DisableAreas = 9;
    string DeepV = 10;
    string ResultOutput = 11;
    SkynetOutput VideoOutput = 12;
}

message SkynetResponseContext {
    
}

message SkynetResponseData {
    
}
